### PR TITLE
Delete deploy

### PR DESCRIPTION
**Pull request by EnriqueL8**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
> /kind code-refactoring

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/odo/cli/component/component.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ func NewCmdComponent(name, fullName string) *cobra.Command {
 	pushCmd := NewCmdPush(PushRecommendedCommandName, odoutil.GetFullName(fullName, PushRecommendedCommandName))
 	updateCmd := NewCmdUpdate(UpdateRecommendedCommandName, odoutil.GetFullName(fullName, UpdateRecommendedCommandName))
 	watchCmd := NewCmdWatch(WatchRecommendedCommandName, odoutil.GetFullName(fullName, WatchRecommendedCommandName))
+	deployDeleteCmd := NewCmdDeployDelete(DeployDeleteRecommendedCommandName, odoutil.GetFullName(fullName, DeployDeleteRecommendedCommandName))
```
Comment (EnriqueL8) [inline pkg/odo/cli/component/component.go]: Should not add deploy delete to component

File: pkg/odo/cli/component/deploy.go [hunk 2]
```diff
@@ -168,6 +178,9 @@ func NewCmdDeploy(name, fullName string) *cobra.Command {
 	//Adding `--project` flag
 	projectCmd.AddProjectFlag(deployCmd)
 
+	//fmt.Println("$$$$$", DeployDeleteRecommendedCommandName, "$$$$$$")
+	deployCmd.AddCommand(deployDeleteCmd)
+	//fmt.Println("$$$$$", deployCmd.Commands()[0], "$$$$$$")
```
Comment (EnriqueL8) [inline pkg/odo/cli/component/deploy.go]: remove println

File: tests/integration/devfile/cmd_devfile_deploy_delete_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	Context("when manifest.yaml is present, but deployment doesn't exist", func() {
+		It("should pass, by deleting the manifest.yaml, but warn the user that deployment doesn't exist ", func() {
+
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, componentName)
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile_deploy.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "manifest.yaml"), filepath.Join(context, ".odo", "manifest.yaml"))
+
+			output := helper.CmdShouldFail("odo", "deploy", "delete")
+			expectedString := "blah"
```
Comment (EnriqueL8) [inline tests/integration/devfile/cmd_devfile_deploy_delete_test.go]: change expected string: "Has deployment was not found" ...

**Comment by CameronMcWilliam** [comment]
Could not delete deployment nodejs-deploy as deployment was not found should be the second test string output, not "blah"

File: pkg/odo/cli/component/component.go [hunk 4]
```diff
@@ -70,7 +71,7 @@ func NewCmdComponent(name, fullName string) *cobra.Command {
 	// add flags from 'get' to component command
 	componentCmd.Flags().AddFlagSet(componentGetCmd.Flags())
 
-	componentCmd.AddCommand(componentGetCmd, createCmd, deleteCmd, describeCmd, deployCmd, linkCmd, unlinkCmd, listCmd, logCmd, pushCmd, updateCmd, watchCmd)
+	componentCmd.AddCommand(componentGetCmd, createCmd, deleteCmd, deployDeleteCmd, describeCmd, deployCmd, linkCmd, unlinkCmd, listCmd, logCmd, pushCmd, updateCmd, watchCmd)
```
Comment (groeges) [inline pkg/odo/cli/component/component.go]: Need to remove `deployDeleteCmd` from here also.

**Review by groeges — APPROVED** [review]
LGTM

**Pull request merged**